examples/custom_methods.rs

```diff
@@ -0,0 +1,65 @@
+#![deny(warnings)]
+use std::net::SocketAddr;
+use std::str::FromStr;
+
+use warp::hyper::StatusCode;
+use warp::{hyper::Method, reject, Filter, Rejection, Reply};
+
+#[derive(Debug)]
+struct MethodError;
+impl reject::Reject for MethodError {}
+
+fn method(name: &str) -> impl Filter<Extract = (), Error = Rejection> + Clone {
```

Comment:
If we assume most methods are going to be known at compile time, we could make this a `&'static str`, and then you don't need to parse it as a `Method`, or clone it, and can just compare `if m == name`, and be done with it.

Reply:
Really good point. Addressed in the new commit.